gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -414,6 +443,18 @@ public boolean isAncestor(IGitCoreCommit presumedAncestor, IGitCoreCommit presum
     });
   }
 
+  @Override
+  public GitCoreRepositoryState deriveRepositoryState() {
+    return Match(jgitRepo.getRepositoryState()).of(
+        Case($(isIn(RepositoryState.CHERRY_PICKING, RepositoryState.CHERRY_PICKING_RESOLVED)),
+            GitCoreRepositoryState.CHERRY_PICK),
+        Case($(isIn(RepositoryState.MERGING, RepositoryState.MERGING_RESOLVED)), GitCoreRepositoryState.MERGING),
+        Case($(isIn(RepositoryState.REBASING, RepositoryState.REBASING_INTERACTIVE, RepositoryState.REBASING_MERGE,
+            RepositoryState.REBASING_REBASING)), GitCoreRepositoryState.REBASING),
+        Case($(isIn(RepositoryState.REVERTING, RepositoryState.REVERTING_RESOLVED)), GitCoreRepositoryState.REVERTING),
+        Case($(), GitCoreRepositoryState.NORMAL));
```

Comment:
Hmm a case for `APPLY` and `BISECTING` is missing (TBH git-machete CLI handles the former but not the latter)